src/commands/addMIConnections/ConnectionsListStep.ts

```diff
@@ -0,0 +1,120 @@
+
+import { type StringDictionary } from "@azure/arm-appservice";
+import { isSettingConvertible } from "@microsoft/vscode-azext-azureappsettings";
+import { AzExtFsExtra, AzureWizardPromptStep, nonNullValue, type IAzureQuickPickItem } from "@microsoft/vscode-azext-utils";
+import * as vscode from 'vscode';
+import { type ILocalSettingsJson } from "../../funcConfig/local.settings";
+import { localize } from "../../localize";
+import { decryptLocalSettings } from "../appSettings/localSettings/decryptLocalSettings";
+import { encryptLocalSettings } from "../appSettings/localSettings/encryptLocalSettings";
+import { getLocalSettingsFile } from "../appSettings/localSettings/getLocalSettingsFile";
+import { type AddMIConnectionsContext } from "./AddMIConnectionsContext";
+
+export interface Connection {
+    name: string;
+    value: string;
+}
+
+export class ConnectionsListStep extends AzureWizardPromptStep<AddMIConnectionsContext> {
+    public async prompt(context: AddMIConnectionsContext): Promise<void> {
+        const picks = await this.getPicks(context);
+
+        if (picks.length === 0) {
+            const noItemFoundMessage: string = localize('noConnectionsFound', 'No connections found in local settings');
+            (await context.ui.showQuickPick(picks, {
+                placeHolder: localize('selectConnections', 'Select the connections you want to add managed identity support for'),
+                suppressPersistence: true,
+                noPicksMessage: noItemFoundMessage
+            }));
+        } else {
+            context.connections = (await context.ui.showQuickPick(picks, {
+                placeHolder: localize('selectConnections', 'Select the connections you want to add managed identity support for'),
+                suppressPersistence: true,
+                canPickMany: true,
+            })).map(item => item.data);
+        }
+    }
+
+    public shouldPrompt(context: AddMIConnectionsContext): boolean {
+        return !context.connections || context.connections.length === 0;
+    }
+
+    private async getPicks(context: AddMIConnectionsContext): Promise<IAzureQuickPickItem<Connection>[]> {
+        if (context.functionapp) {
+            return this.getRemoteQuickPicks(context);
+        } else {
+            return this.getLocalQuickPicks(context);
+        }
+    }
+
+    private async getLocalQuickPicks(context: AddMIConnectionsContext, workspaceFolder?: vscode.WorkspaceFolder): Promise<IAzureQuickPickItem<Connection>[]> {
+        const picks: IAzureQuickPickItem<Connection>[] = [];
+        const message: string = localize('selectLocalSettings', 'Select the local settings to add identity settings for.');
+        const localSettingsPath: string = await getLocalSettingsFile(context, message, workspaceFolder);
+        context.localSettingsPath = localSettingsPath;
+
+        if (await AzExtFsExtra.pathExists(localSettingsPath)) {
```

Comment:
Now that you're using this function, I think you can get rid of the if statement above.